### ISSUE TITLE
Opcode search jmp esp problem ESP ->EIP

### ISSUE THREAD

**Issue by MehmetAliCan10** [eteran/edb-debugger#723]
Hi

Why I can not find Opcode search jmp esp problem ESP ->EIP
I use newest evans edb 

Can you fix it maybe?

**Comment by eteran** [comment]
I suppose that it's possible that there is no `jmp esp` found in the program. Can you share the binary? If so, I can try and see if I can replicate your issue.

**Comment by MehmetAliCan10** [comment]
If I do this:
 edb --run /usr/games/crossfire/bin/crossfire

on OSCP course Opcode Search is Results: 08134597: jmp esp find!

but on newest edb opcode search can not find maybe you can fix it hopefully?
I have search Goto Expression is found=(0x08134597) but not Opcode search....
hope you can fix it ornot?

**Comment by MehmetAliCan10** [comment]
If I do this:
edb --run /usr/games/crossfire/bin/crossfire

on OSCP course Opcode Search is Results: 08134597: jmp esp found!

but on newest edb opcode search can not find maybe you can fix it hopefully?
I have search Goto Expression is found=(0x08134597) but not Opcode search....
hope you can fix it ornot?

**Comment by MehmetAliCan10** [comment]
> I suppose that it's possible that there is no `jmp esp` found in the program. Can you share the binary? If so, I can try and see if I can replicate your issue.


If I do this:
edb --run /usr/games/crossfire/bin/crossfire

on OSCP course Opcode Search is Results: 08134597: jmp esp found!

but on newest edb opcode search can not find maybe you can fix it hopefully?
I have search Goto Expression is found=(0x08134597) but not Opcode search....
hope you can fix it ornot?

**Comment by eteran** [comment]
OK, I'll take a look :-).

**Comment by eteran** [comment]
So, I'm having a hard time replicating this. Can you post what the memory at `0x08134597` is in your binary (or just post a copy of the binary)?

A screenshot of the disassembly starting at the location in question might help as well.

**Comment by MehmetAliCan10** [comment]
I do no know where is memory or binary what you mean...



![image](https://user-images.githubusercontent.com/38558200/67817493-e03ea980-faad-11e9-83cf-7f29890ecf55.png)

**Comment by MehmetAliCan10** [comment]
Here you can also command:
 
cd /usr/games/

wget www.offensive-security.com/crossfire.tar.gz

tar zxpf crossfire.tar.gz

edb --run /usr/games/crossfire/bin/crossfire

**Comment by MehmetAliCan10** [comment]
> So, I'm having a hard time replicating this. Can you post what the memory at `0x08134597` is in your binary (or just post a copy of the binary)?
> 
> A screenshot of the disassembly starting at the location in question might help as well.

I do no know where is memory or binary what you mean...

image

Here you can also command:

cd /usr/games/

wget www.offensive-security.com/crossfire.tar.gz

tar zxpf crossfire.tar.gz

edb --run /usr/games/crossfire/bin/crossfire

**Comment by eteran** [comment]
Appears to work for me.
![edb](https://user-images.githubusercontent.com/571477/67883536-a788e980-fb1a-11e9-9b1c-392ca104dc39.png)

**Comment by MehmetAliCan10** [comment]
huh how is that possible? is that vmware on kali linux 64 bit ?

**Comment by MehmetAliCan10** [comment]
> Appears to work for me.
> ![edb](https://user-images.githubusercontent.com/571477/67883536-a788e980-fb1a-11e9-9b1c-392ca104dc39.png)

huh how is that possible? is that vmware on kali linux 64 bit ?

**Comment by eteran** [comment]
No that's me running the latest version from master in my local build environment.

**Comment by eteran** [comment]
It seems likely to me that Kali Linux has a version of edb that is a bit old, it's been a while since the last release.

**Comment by MehmetAliCan10** [comment]
Can I ask which linux do you use? ubuntu?

**Comment by eteran** [comment]
It shouldn't change anything as far as edb, but I use Gentoo Linux and run edb from compiled sources.

If you want the latest and greatest edb, I highly recommend you install from the git master branch.

**Comment by MehmetAliCan10** [comment]
> It shouldn't change anything as far as edb, but I use Gentoo Linux and run edb from compiled sources.
> 
> If you want the latest and greatest edb, I highly recommend you install from the git master branch.

I do not understend I installed kali linux newest and install newest edb and noting showing opcode search :-( i do no know how did you do please help me thanks....

**Comment by eteran** [comment]
How did you install the "newest"? As I said, there hasn't been a numbered release in a bit, so the "newest release" is not the same as "newest".

If you want to use the actual newest edb, please uninstall the version installed in Kali, remove the configuration file, and follow the directions for compiling from source found in the Wiki.

Assuming Kali is based on Ubuntu, these directions will probably work for you:

https://github.com/eteran/edb-debugger/wiki/Compiling-(Ubuntu)

**Comment by MehmetAliCan10** [comment]
yes thank you now is working. But If I again install this apt-get install edb-debugger will not work... I think apt-get install edb-debugger is old,  you have newest release that why I think :-)

**Comment by eteran** [comment]
Correct, that's what I was trying to explain to you 😂.

Glad you have it working.
